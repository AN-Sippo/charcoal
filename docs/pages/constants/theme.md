# `@charcoal/theme`

charcoal 全体で制約されるテーマ構造の型を提供します。

## インストール

npm

```bash
npm i @charcoal/theme
```

yarn

```bash
yarn add @charcoal/theme
```
