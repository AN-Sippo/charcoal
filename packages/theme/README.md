# `@charcoal/theme`

charcoal 全体で制約されるテーマ構造の型を提供します。

## インストール

npm

```
npm i --save-dev @charcoal/theme
```

yarn

```
yarn add -D @charcoal/theme
```
