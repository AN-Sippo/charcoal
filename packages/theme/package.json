{
  "name": "@charcoal-ui/theme",
  "version": "4.0.0-beta.15",
  "license": "Apache-2.0",
  "main": "./dist/index.cjs.js",
  "module": "./dist/index.esm.js",
  "exports": {
    "types": "./dist/index.d.ts",
    "require": "./dist/index.cjs.js",
    "import": "./dist/index.esm.js",
    "default": "./dist/index.esm.js"
  },
  "types": "./dist/index.d.ts",
  "sideEffects": [
    "*.css"
  ],
  "scripts": {
    "build": "npm-run-all --print-label --parallel 'build:*' --sequential serialize",
    "build:bundle": "FORCE_COLOR=1 tsup",
    "build:dts": "tsc --project tsconfig.build.json --pretty --emitDeclarationOnly",
    "build:copy-css": "cpx 'src/css/**/*.css' dist/css && cpx 'src/json/**/*.json' dist/json",
    "serialize": "node cli/index.js",
    "typecheck": "run-p --print-label 'typecheck:*'",
    "typecheck:main": "tsc --project tsconfig.build.json --pretty --noEmit",
    "typecheck:cli": "tsc --project cli/tsconfig.build.json --noEmit",
    "clean": "rimraf dist .tsbuildinfo",
    "test": "vitest run --passWithNoTests"
  },
  "devDependencies": {
    "cpx": "^1.5.0",
    "npm-run-all": "^4.1.5",
    "rimraf": "^3.0.2",
    "tsup": "^6.5.0",
    "typescript": "^4.9.5",
    "vitest": "^2.0.2"
  },
  "dependencies": {
    "@charcoal-ui/foundation": "^4.0.0-beta.15",
    "@charcoal-ui/utils": "^4.0.0-beta.15",
    "polished": "^4.1.4"
  },
  "files": [
    "src",
    "dist",
    "theme.json"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/pixiv/charcoal.git",
    "directory": "packages/theme"
  },
  "gitHead": "e1ece2e43901ae667afdd5c178040607d939dcd5"
}
