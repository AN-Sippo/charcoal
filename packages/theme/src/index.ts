import { Effect, Effects, OpacityEffect } from './effect'

export default {}

export * from './effect'

export type Material = string
export type GradientMaterial = { color: Material; ratio: number }[]

/**
 * @deprecated
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
type ThemeMaterials = {
  background1: Material
  background2: Material
  surface1: Material
  surface2: Material
  /**
   * Black Fade 20にするとボーダーと共通化できる
   */
  surface3: Material
  surface4: Material
  // NOTE: the current definition of surface5 is invalid
  surface6: Material
  /**
   * サムネイルに被せる色。
   * Lightモードの色が2と共通。コントラストの低下を狙うなら2でいいが、サムネイルで色味を調整しすぎる感がある。
   */
  surface7: Material
  /**
   * 磨りガラス効果を使えれば6と統合できそう
   */
  surface8: Material
  surface9: Material
  link1: Material
  link2: Material
  text1: Material
  /**
   * 1と2は統合できるかもしれない
   */
  text2: Material
  text3: Material
  /**
   * Surface 4に重ねた時に薄すぎる感あり
   */
  text4: Material
  text5: Material
  icon6: Material
  border: Material
}

/**
 * @deprecated
 */
export interface ThemeEffects {
  hover: Effects
  press: Effects
  disabled: Effects
}

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type ThemeEffect = {
  hover: Effect
  press: Effect
}

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type ThemeElementEffect = {
  disabled: OpacityEffect
}

/**
 * @deprecated
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
type ThemeSemantic = {
  brand: Material
}

/**
 * @deprecated Use `ThemeColor` instead
 */
export type ThemeColors = ThemeMaterials &
  ThemeSemantic & {
    premium: Material
  }

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type ThemeColor = {
  transparent: Material
  background1: Material
  background2: Material
  surface1: Material
  surface2: Material
  surface3: Material
  surface4: Material
  surface6: Material
  surface7: Material
  surface8: Material
  surface9: Material
  link1: Material
  link2: Material
  text1: Material
  text2: Material
  text3: Material
  text4: Material
  text5: Material
  icon6: Material
  border: Material
  brand: Material
  assertive: Material
  /**
   * @deprecated
   */
  premium: Material
}

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type ThemeColorGradient = {
  surface4Gradient: GradientMaterial
}

export interface Theme {
  /**
   * @deprecated
   *
   * Use `color` instead.
   */
  materials: ThemeMaterials
  /**
   * @deprecated
   *
   * Use `color` instead.
   */
  semantic: ThemeSemantic
  /**
   * @deprecated Use `color` instead
   */
  colors: ThemeColors
  color: ThemeColor
  gradientColor: ThemeColorGradient
  /**
   * @deprecated Use `effect` instead
   */
  effects: ThemeEffects
  /**
   * effect for color (background or font color)
   */
  effect: ThemeEffect
  /**
   * effect for element its own (opacify element includes descendants)
   */
  elementEffect: ThemeElementEffect
}
