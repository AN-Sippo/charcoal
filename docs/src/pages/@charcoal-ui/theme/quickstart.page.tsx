import Link from 'next/link'
import { ContentRoot } from '../../../components/ContentRoot'

import { InlineCode } from '../../../components/InlineCode'
import { SSRHighlight } from '../../../components/SSRHighlight'

export default function InstallPage() {
  return (
    <ContentRoot>
      <h1>@charcoal-ui/theme クイックスタート</h1>
      <Link href="https://www.npmjs.com/package/@charcoal-ui/theme">
        <img
          alt="npm-badge"
          src="https://img.shields.io/npm/v/@charcoal-ui/theme?label=%40charcoal-ui%2Ftheme&style=flat-square&logo=npm"
        />
      </Link>
      <h2>インストール</h2>
      <h3>npm</h3>
      <SSRHighlight
        code="npm install --save-dev @charcoal-ui/theme"
        lang="shell"
      />
      <h3>yarn</h3>
      <SSRHighlight code="yarn add -D @charcoal-ui/theme" lang="shell" />

      <hr />

      <h2>概要</h2>
      <p>
        <InlineCode>@charcoal-ui/theme</InlineCode>
        は、全体で使われるテーマ型と、デフォルトのライト・ダークテーマを提供します。
      </p>

      <h2>使い方</h2>
      <p>
        <InlineCode>styled-components</InlineCode>と合わせて使用できます。
      </p>
      <p>
        デフォルトのテーマとして提供している<InlineCode>light</InlineCode>、
        <InlineCode>dark</InlineCode>をそのまま使うこともできます。
      </p>
      <SSRHighlight code={sampleCode} lang="typescript" />
    </ContentRoot>
  )
}

const sampleCode = `
  import { light, dark } from '@charcoal-ui/theme'
  import { ThemeProvider } from 'styled-components'
  import { FC, ReactNode } from 'react'

  const LightExample: FC<{ children: ReactNode }> = ({ children }) => (
    <ThemeProvider theme={light}>
      {children}
    </ThemeProvider>
  )

  const DarkExample: FC<{ children: ReactNode }> = ({ children }) => (
    <ThemeProvider theme={dark}>
      {children}
    </ThemeProvider>
  )
`
