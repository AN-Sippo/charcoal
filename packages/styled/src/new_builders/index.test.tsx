import { light } from '@charcoal-ui/theme'
import 'jest-styled-components'

import renderder from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { myTheme } from '../storyHelper'
import { Example, TailwindLike } from './index.story'

function render(children: JSX.Element) {
  return renderder
    .create(<ThemeProvider theme={myTheme(light)}>{children}</ThemeProvider>)
    .toJSON()
}

describe('Story', () => {
  test('<Example />', () => {
    expect(render(<Example />)).toMatchSnapshot()
  })

  test('<TailwindLike />', () => {
    expect(render(<TailwindLike />)).toMatchSnapshot()
  })
})
