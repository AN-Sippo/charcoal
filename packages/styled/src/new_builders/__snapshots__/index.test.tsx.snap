// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story <Example /> 1`] = `
.c1 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text4);
  display: flow-root;
  font-size: 32px;
  line-height: 40px;
  font-family: monospace;
  margin-top: 24px;
  margin-bottom: 24px;
  padding-left: 64px;
  padding-top: 64px;
}

.c1::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c1::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c0 {
  background-color: var(--charcoal-surface2);
  color: var(--charcoal-text2);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  padding: 36px 40px 36px 40px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c2 {
  background-color: var(--charcoal-surface2);
  padding: 40px;
}

.c3 {
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  display: flow-root;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  height: 40px;
}

.c3::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c3::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c4 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text1);
  display: flow-root;
  font-size: 16px;
  line-height: 24px;
  width: 392px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c4::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c4::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c5 {
  height: 64px;
}

.c6 {
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  margin-top: 64px;
  margin-bottom: 24px;
  padding: 60px 64px 60px 64px;
  -webkit-transition: 0.2s background-color,0.2s color;
  transition: 0.2s background-color,0.2s color;
}

.c6:hover:not(:disabled):not([aria-disabled]),
.c6[aria-disabled=false] {
  background-color: var(--charcoal-surface4-hover);
}

.c6:active:not(:disabled):not([aria-disabled]),
.c6[aria-disabled=false] {
  background-color: var(--charcoal-surface4-press);
}

.c6:hover:not(:disabled):not([aria-disabled]),
.c6[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c6:active:not(:disabled):not([aria-disabled]),
.c6[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c8 {
  background-color: var(--charcoal-surface3);
  display: flow-root;
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  margin-top: 24px;
  margin-bottom: 24px;
  padding: 24px;
}

.c8::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c8::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c7 {
  background-color: var(--charcoal-surface3);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  padding: 20px 24px 20px 24px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c9 {
  background-color: var(--charcoal-surface3);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  padding: 24px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c10 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  display: flow-root;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  width: 288px;
  height: 64px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c10::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c10::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c11 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  gap: 16px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c12 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  cursor: pointer;
  background-color: var(--charcoal-brand);
  color: var(--charcoal-text5);
  display: flow-root;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  height: 40px;
  padding-left: 24px;
  padding-right: 24px;
  border-radius: 99999px;
  -webkit-transition: 0.2s background-color,0.2s color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s color,0.2s box-shadow;
}

.c12:focus {
  outline: none;
}

.c12:hover:not(:disabled):not([aria-disabled]),
.c12[aria-disabled=false] {
  background-color: var(--charcoal-brand-hover);
}

.c12:active:not(:disabled):not([aria-disabled]),
.c12[aria-disabled=false] {
  background-color: var(--charcoal-brand-press);
}

.c12:hover:not(:disabled):not([aria-disabled]),
.c12[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c12:active:not(:disabled):not([aria-disabled]),
.c12[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c12::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c12::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c12:disabled {
  opacity: 0.32;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c12:disabled,
.c12[aria-disabled]:not([aria-disabled=false]) {
  cursor: unset;
}

.c13 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  height: 40px;
  width: 184px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 4px;
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  -webkit-transition: 0.2s background-color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s box-shadow;
}

.c13:focus {
  outline: none;
}

.c13:hover:not(:disabled):not([aria-disabled]),
.c13[aria-disabled=false] {
  background-color: var(--charcoal-surface3-hover);
}

.c13:disabled {
  opacity: 0.32;
}

.c13:focus {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c14 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  height: 40px;
  width: 184px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 0px;
  padding-bottom: 0px;
  border-radius: 4px;
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  outline: none;
  box-shadow: 0 0 0 4px rgba(255,43,0,0.32);
  -webkit-transition: 0.2s background-color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s box-shadow;
}

.c14:focus {
  outline: none;
}

.c14:hover:not(:disabled):not([aria-disabled]),
.c14[aria-disabled=false] {
  background-color: var(--charcoal-surface3-hover);
}

.c14:disabled {
  opacity: 0.32;
}

.c14:focus {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c15 {
  background-color: var(--charcoal-background1);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  width: 392px;
  margin-top: 40px;
  margin-bottom: 40px;
  padding: 20px 24px 20px 24px;
  border: 1px solid var(--charcoal-border-default);
  border-radius: 8px;
}

.c16 {
  background-color: #d1ff1a;
  background-image: linear-gradient(to right,#d1ff1a 0%,#1ad1ff 100%);
  color: var(--charcoal-text5);
  width: 184px;
  padding: 36px 40px 36px 40px;
  margin-top: 24px;
  margin-bottom: 24px;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  border-radius: 8px;
  -webkit-transition: 0.2s background-color,0.2s color;
  transition: 0.2s background-color,0.2s color;
  text-align: center;
}

.c16:hover:not(:disabled):not([aria-disabled]),
.c16[aria-disabled=false] {
  background-color: var(--charcoal-background1-hover);
}

.c16:active:not(:disabled):not([aria-disabled]),
.c16[aria-disabled=false] {
  background-color: var(--charcoal-background1-press);
}

.c16:hover:not(:disabled):not([aria-disabled]),
.c16[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c16:active:not(:disabled):not([aria-disabled]),
.c16[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c17 {
  background-color: #d1ff1a;
  background-image: linear-gradient(to right,#d1ff1a 0%,#1ad1ff 100%);
  color: var(--charcoal-text5);
  width: 184px;
  margin-top: 24px;
  margin-bottom: 24px;
  display: flow-root;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  border-radius: 8px;
  -webkit-transition: 0.2s background-color,0.2s color;
  transition: 0.2s background-color,0.2s color;
  padding: 40px;
  text-align: center;
}

.c17:hover:not(:disabled):not([aria-disabled]),
.c17[aria-disabled=false] {
  background-color: var(--charcoal-background1-hover);
}

.c17:active:not(:disabled):not([aria-disabled]),
.c17[aria-disabled=false] {
  background-color: var(--charcoal-background1-press);
}

.c17:hover:not(:disabled):not([aria-disabled]),
.c17[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c17:active:not(:disabled):not([aria-disabled]),
.c17[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c17::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c17::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c18 {
  width: 288px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c19 {
  display: inline-block;
  width: 100%;
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  padding: 16px;
}

.c20 {
  --charcoal-text1: #ffff00;
  --charcoal-text1-hover: #f5f500;
  --charcoal-text1-press: #d6d600;
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text1);
}

<div
  className=""
>
  <div
    className="c0"
  >
    Sample
  </div>
  <div
    className="c1"
  >
    Left Top Padding
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      Nested
    </div>
  </div>
  <div
    className="c4"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Autem dolores, recusandae quidem mollitia eum non vel architecto possimus repudiandae quis molestias neque facilis rem dolorum voluptatem impedit nemo praesentium voluptas.
  </div>
  <div
    className="c4 c5"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Autem dolores, recusandae quidem mollitia eum non vel architecto possimus repudiandae quis molestias neque facilis rem dolorum voluptatem impedit nemo praesentium voluptas.
  </div>
  <div
    className="c6"
  >
    With effects
  </div>
  <div
    className="c7"
  >
    Without half-leading
  </div>
  <div
    className="c8"
  >
    Without half-leading (No optimization)
  </div>
  <div
    className="c9"
  >
    With half-leading
  </div>
  <div
    className="c10"
  >
    Fixed size box
  </div>
  <div
    className="c11"
  >
    <button
      className="c12"
    >
      Button
    </button>
    <button
      className="c12"
      disabled={true}
    >
      Disabled
    </button>
  </div>
  <div
    className="c11"
  >
    <input
      className="c13"
      type="text"
      value="text field"
    />
    <input
      className="c13"
      disabled={true}
      type="text"
      value="disabled"
    />
    <input
      className="c14"
      type="text"
      value="invalid"
    />
  </div>
  <div
    className="c15"
  >
    Border
  </div>
  <div
    className="c16"
  >
    Gradient
  </div>
  <div
    className="c17"
  >
    Gradient (Warning)
  </div>
  <div
    css="
      display: flex;
    "
  >
    <div
      css={
        Array [
          Array [
            "
            ",
            "color",
            ": var(--charcoal-",
            "text5",
            ");
          ",
          ],
          Array [
            "
          :",
            "hover",
            ":not(:disabled):not([aria-disabled]),&[aria-disabled=false]  {
            ",
            "color",
            ": var(--charcoal-",
            "text5",
            "-",
            "hover",
            ");
          }
        ",
          ],
          Array [
            "
          :",
            "active",
            ":not(:disabled):not([aria-disabled]),&[aria-disabled=false]  {
            ",
            "color",
            ": var(--charcoal-",
            "text5",
            "-",
            "press",
            ");
          }
        ",
          ],
          Array [
            "
            font-size: ",
            "14",
            "px;
            line-height: ",
            "22",
            "px;
          ",
          ],
          Array [
            "
                font-weight: bold;
              ",
          ],
          Array [
            "
                  ",
            "padding",
            ": ",
            "12",
            "px ",
            "16",
            "px ",
            "12",
            "px ",
            "16",
            "px;
                ",
          ],
          Array [
            "
              border-radius: ",
            "8",
            "px;
            ",
          ],
          Array [
            "
    transition: ",
            "0.2s color",
            ";
  ",
          ],
        ]
      }
    >
      Tailwind like
    </div>
  </div>
  <div
    className="c18"
  >
    <div
      className="c19"
    >
      Full width
    </div>
  </div>
  <div
    className="c20"
  >
    This is actually text1 !
  </div>
</div>
`;

exports[`Story <TailwindLike /> 1`] = `
<div
  css="
      display: flex;
    "
>
  <div
    css={
      Array [
        Array [
          "
            ",
          "color",
          ": var(--charcoal-",
          "text5",
          ");
          ",
        ],
        Array [
          "
          :",
          "hover",
          ":not(:disabled):not([aria-disabled]),&[aria-disabled=false]  {
            ",
          "color",
          ": var(--charcoal-",
          "text5",
          "-",
          "hover",
          ");
          }
        ",
        ],
        Array [
          "
          :",
          "active",
          ":not(:disabled):not([aria-disabled]),&[aria-disabled=false]  {
            ",
          "color",
          ": var(--charcoal-",
          "text5",
          "-",
          "press",
          ");
          }
        ",
        ],
        Array [
          "
            font-size: ",
          "14",
          "px;
            line-height: ",
          "22",
          "px;
          ",
        ],
        Array [
          "
                font-weight: bold;
              ",
        ],
        Array [
          "
                  ",
          "padding",
          ": ",
          "12",
          "px ",
          "16",
          "px ",
          "12",
          "px ",
          "16",
          "px;
                ",
        ],
        Array [
          "
              border-radius: ",
          "8",
          "px;
            ",
        ],
        Array [
          "
    transition: ",
          "0.2s color",
          ";
  ",
        ],
      ]
    }
  >
    Tailwind like
  </div>
</div>
`;
